File: packages/expo-symbols/ios/SymbolModule.swift
import ExpoModulesCore

public class SymbolModule: Module {
  public func definition() -> ModuleDefinition {
    Name("SymbolModule")

    View(SymbolView.self) {
      Prop("name") { (view, name: String) in
        view.name = name
      }

      Prop("type") { (view, type: SymbolType?) in
        view.symbolType = type ?? .monochrome
      }

      Prop("scale") { (view, scale: SymbolScale?) in
        view.scale = scale?.imageSymbolScale() ?? .unspecified
      }

      Prop("tintColor") { (view, color: UIColor?) in
        view.tint = color
      }

      Prop("animated") { (view, animated: Bool?) in
        view.animated = animated ?? false
      }

      Prop("weight") { (view, weight: SymbolWeight?) in
        view.weight = weight?.imageSymbolWeight() ?? .regular
      }

      Prop("colors") { (view, color: [UIColor]?) in
        view.palette = color ?? []
      }

      Prop("resizeMode") { (view, resizeMode: SymbolContentMode?) in
        view.imageContentMode = resizeMode?.toContentMode() ?? .scaleAspectFit
      }

      Prop("animationSpec") { (view, spec: AnimationSpec?) in
        view.animationSpec = spec
      }

      OnViewDidUpdateProps { view in
        view.reloadSymbol()
      }
    }
  }
}

File: packages/expo-symbols/android/src/main/java/expo/modules/symbols/SymbolModule.kt
package expo.modules.symbols

import android.graphics.Color
import android.os.Build
import androidx.annotation.RequiresApi
import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class SymbolModule : Module() {

  override fun definition() = ModuleDefinition {
    Name("ExpoSymbols")

    View(SymbolView::class) {
      Prop("name") { view, name: String ->
        view.name.value = name
      }

      Prop("tintColor") { view, color: Color? ->
        @RequiresApi(Build.VERSION_CODES.O)
        view.tint.value = androidx.compose.ui.graphics.Color(color?.toArgb() ?: Color.WHITE)
      }
    }
  }
}
